File: src/main/java/assignment2/SyntaxErrorException.java
package assignment2;

/**
 * Exception thrown to indicate a syntax error at a specific line in the source code.
 */
public class SyntaxErrorException extends CompilerException {

    /**
     * Constructs a new SyntaxErrorException with the specified detail message and line number.
     *
     * @param message the detail message
     * @param line    the line number where the syntax error occurred
     */
    public SyntaxErrorException(String message, int line) {
        super(message, line);
    }

    /**
     * Returns a string representation of the syntax error exception.
     *
     * @return a string describing the syntax error and its line number
     */
    @Override
    public String toString() {
        return "Syntax error at line " + getLine() + ": " + getMessage();
    }
}

File: src/main/scala/assignment3/SyntaxErrorException.scala
package assignment3

/** Exception thrown to indicate a syntax error at a specific location in the source code (Scala port).
  * @deprecated Use SyntaxDiag instead. Will be removed in 3.0.
  */
@deprecated("Use SyntaxDiag instead", "2.0")
class SyntaxErrorException(message: String, line: Int, column: Int) extends SourcePositionedException(message, line, column) {
  def this(message: String, line: Int) = this(message, line, -1)
}
